engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -157,7 +157,11 @@ trait CromwellApiService extends HttpService with PerRequestCreator {
           } else Success(workflowInputs map { workflowInputSet =>
             mergeMaps(Seq(Option(workflowInputSet)) ++ sortedInputAuxes).toString
           })
-          wfInputs.map(_.map(x => WorkflowSourceFilesCollection(wdlSource, x, workflowOptions.getOrElse("{}"), labels.getOrElse("{}"), wdlDependencies)))
+          val wfOptions = WorkflowOptions.fromJsonString(workflowOptions.getOrElse("{}")) match {
+            case Failure(err) => Failure(new IllegalArgumentException(s"Invalid workflow options provided: ${err.getMessage}"))
```

Comment:
Could we make `err` a `causedBy` of the `IllegalArgumentException`? Oftentimes the message string for those is pretty useless without the extra context.